src/cargo/ops/cargo_add/mod.rs

```diff
@@ -349,10 +369,15 @@ fn resolve_dependency(
             }
             selected
         } else {
-            let mut source = crate::sources::PathSource::new(&path, src.source_id()?, gctx);
+            let mut source = crate::sources::PathSource::new(&src.path, src.source_id()?, gctx);
             let package = source.root_package()?;
             Dependency::from(package.summary())
         };
+        if let Some(selected_source) = selected.source.as_mut() {
+            if let Source::Path(selected_source) = selected_source {
+                selected_source.base_name_and_value = base_name_and_value;
+            }
+        }
```

Comment:
I feel like this should be part of the creation of `src`.  Thst leaves `Dependency::from(package.summary())` but that should then be handled within that branch

Reply:
Unfortunately, this doesn't work: `select_package` creates a new `Dependency` object without re-using the original `Source` information, so the `base` information is lost.

Reply:
two-phase initialization is not great and can lead to its own bugs, in addition to being messy.

I think I'd prefer us to handle it in `select_package`

Reply:
Ok, I pushed this down into the `Summary` to `Dependency` conversion.